pkg/discovery/network/network.go

```diff
@@ -95,24 +95,24 @@ func Discover(dynClient dynamic.Interface, clientSet kubernetes.Interface, submC
 	}
 }
 
-func networkPluginsDiscovery(dynClient dynamic.Interface, clientSet kubernetes.Interface) (*ClusterNetwork, error) {
+func networkPluginsDiscovery(dynClient dynamic.Interface, clientSet kubernetes.Interface) *ClusterNetwork {
 
 	osClusterNet, err := discoverOpenShift4Network(dynClient)
 	if err == nil && osClusterNet != nil {
-		return osClusterNet, nil
+		return osClusterNet
 	}
 
 	weaveClusterNet, err := discoverWeaveNetwork(clientSet)
 	if err == nil && weaveClusterNet != nil {
-		return weaveClusterNet, nil
+		return weaveClusterNet
 	}
 
 	canalClusterNet, err := discoverCanalFlannelNetwork(clientSet)
 	if err == nil && canalClusterNet != nil {
-		return canalClusterNet, nil
+		return canalClusterNet
 	}
 
-	return nil, nil
+	return nil
```

Comment:
Given that the code always returned a nil error, it seems reasonable to remove the return value. OTOH, if an error occurred that prevented the discovery of a technology, eg a k8s API server failure, then I think it should return the error and the caller should bail. So a non-nil error means no determination could be made, otherwise a nil `ClusterNetwork` means it definitively wasn't found. But I think that's outside the scope of this PR - let's just fix the linting errors as simply as we can for now.

Reply:
I pushed https://github.com/submariner-io/submariner-operator/pull/581 to refactor the error handling which will also fix the linting error.

Reply:
Awesome, thanks @tpantelis! I was nervous we'd end up losing this great feedback about the better solution. I'll enable the linter on top of that better-fix.